asr-speaker-diarization.ipynb

```diff
@@ -0,0 +1,236 @@
+   "metadata": {},
+   "source": [
+    "## NVIDIA Riva Overview\n",
+    "\n",
+    "NVIDIA Riva is a GPU-accelerated SDK for building Speech AI applications that are customized for your use case and deliver real-time performance. <br/>\n",
+    "Riva offers a rich set of speech and natural language understanding services such as:\n",
+    "\n",
+    "- Automated speech recognition (ASR)\n",
+    "- Text-to-Speech synthesis (TTS)\n",
+    "- A collection of natural language processing (NLP) services, such as named entity recognition (NER), punctuation, intent classification.\n",
+    "\n",
+    "In this tutorial, we will show how to use Speaker Diarization feature with Riva ASR to get the transcript with each word tagged with the id of the speaker who has spoken that word. <br> \n",
+    "To understand the basics of Riva ASR APIs, refer to [Getting started with Riva ASR in Python](https://github.com/nvidia-riva/tutorials/tree/stable/asr-python-basics.ipynb). <br>\n",
+    "\n",
+    "For more information about Riva, refer to the [Riva developer documentation](https://developer.nvidia.com/riva)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "27f37136",
+   "metadata": {},
+   "source": [
+    "## Speaker Diarization with Riva ASR APIs\n",
+    "\n",
+    "[Speaker Diarization](https://en.wikipedia.org/wiki/Speaker_diarisation) is the process of segmenting audio recordings by speaker labels and aims to answer the question “who spoke when?”. Riva ASR supports speaker diarization which can be enabled by passing [SpeakerDiarizationConfig](https://docs.nvidia.com/deeplearning/riva/user-guide/docs/reference/protos/protos.html#_CPPv424SpeakerDiarizationConfig) with `enable_speaker_diarization` set to `True`. Riva Speaker diarization segments input audio, extracts speaker embeddings of the segments, counts the number of speakers and then assigns each audio segment a corresponding speaker tag. When Speaker Diarization is enabled, Riva ASR returns ASR transcript to the client, along with a speaker tag for each word in the transcript. Speaker Diarization is language agnostic and can work with any language supported by Riva ASR.\n",
+    "\n",
+    "### Note:\n",
+    "Speaker Diarization is only supported with Riva ASR Offline API. Speaker diarization is an alpha release and will increase ASR latency once the feature is enabled. Please refer [Performance](https://docs.nvidia.com/deeplearning/riva/user-guide/docs/asr/asr-performance.html) for details."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3304f0f0",
+   "metadata": {},
+   "source": [
+    "# Requirements and setup\n",
+    "\n",
+    "1. Enable Speaker Diarization model  \n",
+    "Speaker Diarization is an optional model, so it needs to be enabled in `config.sh` by unncommenting the line containing `rmir_diarizer_offline` word. Since Speaker Diarization works only with Riva ASR Offline API, make sure that the Offline ASR model is also enabled in `config.sh`.\n",
```

Comment:
```suggestion
    "Speaker Diarization is an optional model, so it needs to be enabled in `config.sh` by unncommenting the line containing the `rmir_diarizer_offline` word. Since speaker diarization works only with Riva ASR offline API, make sure that the offline ASR model is also enabled in `config.sh`.\n",
```